lib/browser.ts

```diff
@@ -837,79 +848,85 @@ export class ProtractorBrowser extends AbstractExtendedWebDriver {
       return 'Protractor.get(' + destination + ') - ' + str;
     };
 
+    let retPromise: wdpromise.Promise<any>;
+    let then = (fn: () => wdpromise.Promise<any>) => {
+      if (retPromise) {
+        return (retPromise = retPromise.then(fn));
+      } else {
+        return (retPromise = fn());
+      }
+    };
+
     if (this.bpClient) {
```

Comment:
Move this below the this.ignoreSynchronization check, I think that'll fix the test on Circle.